### PR TITLE
Make patching API public

### PR DESCRIPTION
**Pull request by rsalvador**
to be used by bazel’s error prone plugin: https://github.com/bazelbuild/bazel/pull/21640

### CODE REVIEW

**Comment by msridhar** [comment]
@cushon this change is being successfully used at Uber to enable Error Prone auto-patching to work with Bazel, along with https://github.com/bazelbuild/bazel/pull/21640.  I have a follow-on Error Prone fix that makes auto-patching also work for plug-in checks, which I'm happy to contribute as a PR as well (though it makes more sense to land after this one).

Independent of the Bazel change, this PR seems straightforward and could be helpful for any build system trying to integrate Error Prone.  So, it'd be great if someone from the EP team could review it sometime.  Thanks!

File: check_api/src/main/java/com/google/errorprone/ErrorProneAnalyzer.java [hunk 1]
```diff
@@ -57,6 +61,73 @@ public class ErrorProneAnalyzer implements TaskListener {
...
+    refactoringCollection[0] = RefactoringCollection.refactor(epOptions.patchingOptions(), context);
+
+    // Refaster refactorer or using builtin checks
+    CodeTransformer codeTransformer =
+        epOptions
+            .patchingOptions()
+            .customRefactorer()
+            .or(
+                () -> {
+                  ScannerSupplier toUse =
```
Comment (cushon) [inline check_api/src/main/java/com/google/errorprone/ErrorProneAnalyzer.java:81]: This un-does some changes from https://github.com/google/error-prone/commit/9da2d5580e3939f97ef2e91278b330a56b5ed1fe compared to the original code, I wonder if that was just an oversight when picking up 9da2d5580e3939f97ef2e91278b330a56b5ed1fe?

Comment (rsalvador) [reply check_api/src/main/java/com/google/errorprone/ErrorProneAnalyzer.java]: Yes, was an oversight, re-pushed the changes using the latest code.

**Review by cushon — APPROVED** [review]

**Comment by msridhar** [comment]
@cushon are any other steps required for this one to land?  Thanks for the help!

**Pull request closed without merging**